src/main/java/modbus/ModbusConnection.java

```diff
@@ -0,0 +1,366 @@
+			}
+			act = getAddDeviceAction();
+			anode = node.getChild(getAddDeviceActionName());
+			if (anode == null) {
+				node.createChild(getAddDeviceActionName()).setAction(act).build().setSerializable(false);
+			} else {
+				anode.setAction(act);
+			}
+		}
+	}
+
+	class StopHandler implements Handler<ActionResult> {
+		public void handle(ActionResult event) {
+			stop();
+		}
+	}
+
+	class RestartHandler implements Handler<ActionResult> {
+		public void handle(ActionResult event) {
+			stop();
+			restoreLastSession();
+		}
+	}
+
+	public ScheduledThreadPoolExecutor getDaemonThreadPool() {
+		return stpe;
+	}
+
+	public Action getRemoveAction() {
+		Action act = new Action(Permission.READ, new Handler<ActionResult>() {
+			public void handle(ActionResult event) {
+				remove();
+			}
+		});
+
+		return act;
+	}
+
+	ModbusLink getLink() {
+		return this.link;
+	}
+
+	Action getAddDeviceAction() {
+		Action act = new Action(Permission.READ, getAddDeviceActionHandler());
+		act.addParameter(new Parameter(ATTR_SLAVE_NAME, ValueType.STRING));
+		act.addParameter(new Parameter(ATTR_SLAVE_ID, ValueType.NUMBER, new Value(1)));
+		act.addParameter(new Parameter(ATTR_POLLING_INTERVAL, ValueType.NUMBER, new Value(5)));
+		act.addParameter(new Parameter(ATTR_ZERO_ON_FAILED_POLL, ValueType.BOOL, new Value(false)));
+		act.addParameter(new Parameter(ATTR_USE_BATCH_POLLING, ValueType.BOOL, new Value(true)));
+		act.addParameter(new Parameter(ATTR_CONTIGUOUS_BATCH_REQUEST_ONLY, ValueType.BOOL, new Value(false)));
+
+		return act;
+
+	}
+
+	void makeStartAction() {
+		Action act = new Action(Permission.READ, new Handler<ActionResult>() {
+			public void handle(ActionResult event) {
+				master = getMaster();
+				if (null != master) {
+					checkConnection();
+					node.removeChild("start");
+					makeStopAction();
+				}
+
+			}
+		});
+		Node anode = node.getChild("start");
+		if (anode == null)
+			node.createChild("start").setAction(act).build().setSerializable(false);
+		else
+			anode.setAction(act);
+	}
+
+	void makeStopAction() {
+		Action act = new Action(Permission.READ, new Handler<ActionResult>() {
+			public void handle(ActionResult event) {
+				if (master != null) {
+					master.destroy();
+					link.masters.remove(master);
+					master = null;
+				}
+				node.removeChild("stop");
+				statnode.setValue(new Value("Stopped"));
+				makeStartAction();
+			}
+		});
+		Node anode = node.getChild("stop");
+		if (anode == null)
+			node.createChild("stop").setAction(act).build().setSerializable(false);
+		else
+			anode.setAction(act);
+	}
+
+	void checkConnection() {
+		boolean connected = false;
+		if (master != null) {
+			try {
+				LOGGER.debug("pinging device to test connectivity");
+				connected = master.testSlaveNode(node.getAttribute("slave id").getNumber().intValue());
+			} catch (Exception e) {
+				LOGGER.debug("error during device ping: ", e);
+			}
+			if (connected)
+				statnode.setValue(new Value("Ready"));
```

Comment:
Should that be a constant or an Enum value? Looks like it's referenced a few times in clear text like this.